### PR TITLE
[cleanup] mostly constification and other profile code straightening

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This is the last preparatory PR before making the profile reentrant. Mostly constification. Some people reallly seem to dislike const. Not much to say, really. Ultimately this should lead to a dynamic profile on mobile (we then can have multiple profile views at the same time) and less insane printing.

### CODE REVIEW

File: profile-widget/profilewidget2.cpp [hunk 1]
```diff
@@ -575,14 +575,7 @@ void ProfileWidget2::plotDive(const struct dive *d, bool force, bool doClearPict
 	if (currentState == EMPTY)
 		setProfileState();
 
-	// next get the dive computer structure - if there are no samples
-	// let's create a fake profile that's somewhat reasonable for the
-	// data that we have
 	struct divecomputer *currentdc = select_dc(&displayed_dive);
-	Q_ASSERT(currentdc);
-	if (!currentdc || !currentdc->samples)
-		fake_dc(currentdc);
```
Comment (dirkhh) [inline profile-widget/profilewidget2.cpp:586]: would it make sense to print out a `qWarning()` if there aren't any samples?
I'm sure you checked all the paths, but still... or am I too paranoid?

**Review by dirkhh — COMMENTED** [review]
all the constifications (is that a word??) are great. I am mildly worried about the `fake_dc()` thingy... what are your thoughts on my suggestion?

**Comment by bstoeger** [comment]
Forget this PR for now - it is broken.

**Comment by bstoeger** [comment]
Fixed the problem - time to reopen.

**Comment by dirkhh** [comment]
Interestingly I don't see a new commit pushed... so the problem wasn't in the code?

**Review by dirkhh — APPROVED** [review]
I continue to think this is great.
The one fairly invasive change is of course the initial phase of dive planning - but if that were to break something that should be pretty obvious, and in my testing it seemed to work fine.

**Pull request merged**